app/code/community/Twispay/Tpay/controllers/PaymentController.php

```diff
@@ -1,362 +1,280 @@
+                   ];
+      }
+
+      /* Construct the backUrl. */
+      $backUrl =  Mage::getBaseUrl() . "tpay/payment/response/";
+
+      /* Calculate the order amount. */
+      $amount = $order->getBaseGrandTotal();
+      $index = strpos($amount, '.');
+      if(FALSE !== $index){
+        $amount = substr($amount, 0, $index + 3);
+      }
+
+      /* Build the data object to be posted to Twispay. */
+      $orderData = [ 'siteId' => $siteId
+                   , 'customer' => $customer
+                   , 'order' => [ 'orderId' => $orderId
+                                , 'type' => 'purchase'
+                                , 'amount' => $amount
+                                , 'currency' => $order->getOrderCurrencyCode()
+                                , 'items' => $items
+                                /* , 'tags' => [] */
+                                /* , 'intervalType' => '' */
+                                /* , 'intervalValue' => 1 */
+                                /* , 'trialAmount' => 1 */
+                                /* , 'firstBillDate' => '' */
+                                /* , 'level3Type' => '', */
+                                /* , 'level3Airline' => [ 'ticketNumber' => '' */
+                                /*                      , 'passengerName' => '' */
+                                /*                      , 'flightNumber' => '' */
+                                /*                      , 'departureDate' => '' */
+                                /*                      , 'departureAirportCode' => '' */
+                                /*                      , 'arrivalAirportCode' => '' */
+                                /*                      , 'carrierCode' => '' */
+                                /*                      , 'travelAgencyCode' => '' */
+                                /*                      , 'travelAgencyName' => ''] */
+                                ]
+                   , 'cardTransactionMode' => 'authAndCapture'
+                   /* , 'cardId' => 0 */
+                   , 'invoiceEmail' => ''
+                   , 'backUrl' => $backUrl
+                   /* , 'customData' => [] */
+      ];
+
+      Mage::Log(__FUNCTION__ . ': orderData=' . print_r($orderData, TRUE), Zend_Log::DEBUG, $this->logFileName);
+
+      /* Encode the data and calculate the checksum. */
+      $base64JsonRequest = Mage::helper('tpay')->getBase64JsonRequest($orderData);
+      Mage::Log(__FUNCTION__ . ': base64JsonRequest=' . $base64JsonRequest, Zend_Log::DEBUG, $this->logFileName);
+      $base64Checksum = Mage::helper('tpay')->getBase64Checksum($orderData, $apiKey);
+      Mage::Log(__FUNCTION__ . ': base64Checksum=' . $base64Checksum, Zend_Log::DEBUG, $this->logFileName);
+
+      $link = $url;
+      $payment = $order->getPayment();
+      $payment->setTransactionId(time()); // Make it unique.
```

Comment:
There could be two (or more) requests processed in the same second by the server, using just `time()` call does not ensure uniqueness. Previous version of code also appended `$orderId` ...

Reply:
I can add append or prepend the $orderId.

Can you the format I want to use?

The previous version used only the time (please see https://github.com/Twispay/magento1.x/blob/master/app/code/community/Twispay/Tpay/controllers/PaymentController.php line 64 + line 127).
Maybe you are referring to the "identifier" field that is set to the Twispay server which was calculated like this ($data['identifier'] = "_".$rmTranid;) and is now calculated like following:
'identifier' => (0 == $billingAddress->getCustomerId()) ? ('_' . $orderId . '_' . date('YmdHis')) : ('_' . $billingAddress->getCustomerId())

Reply:
Yes, sorry my mistake. The value of `identifier` it's ok. If the value of `$payment->setTransactionId(...)` need to be unique, we should use something more than just calling `time()`, prepending `orderId` should do it.

Reply:
I prepended the $orderId. The transaction ID is now constructed like following:
 $payment->setTransactionId($orderId . '_' . time());